src/components/organisms/modal/modal.js

```diff
@@ -0,0 +1,85 @@
+import { useState } from "react";
+import { Input, Button } from "../../";
+import { AiFillDelete } from "react-icons/ai";
+import { MdOutlineAddCircle } from "react-icons/md";
+import "./modal.css";
+import {
+  generatePlaylist,
+  removePlaylist,
+  addVideoToPlaylist,
+  useAuth,
+  useFeature,
+} from "../../../helpers";
+export function Modal({ setVisible, video }) {
+  const [text, setText] = useState("");
+  const { dispatchFeature, playlists } = useFeature();
+  const { token } = useAuth();
+  const addPlaylist = () => {
+    text.length
+      ? generatePlaylist(text, dispatchFeature, token)
+      : alert("Mera naam kya hai? Mujhe naam do");
+    setText("");
+  };
+
+  const deletePlaylist = (id) => removePlaylist(dispatchFeature, token, id);
+  const addToPlaylist = (id) =>
+    addVideoToPlaylist(dispatchFeature, token, id, video);
+  return (
+    <div className="modalBackground flex justify-center align-center">
+      <div className="modalContainer flex flex-column text-white">
+        <h1
+          className="text-white cross-for-chips text-center"
+          onClick={() => setVisible(false)}
+        >
+          Close
+        </h1>
+        <section
+          className="flex align-center justify-space-between"
+          style={{ marginBottom: "1rem" }}
+        >
+          <Input
+            inputClass="input-text md"
+            inputName="playlist"
+            inputPlaceHolder="create playlist..."
+            inputValue={text}
+            inputFunc={(e) => setText(e.target.value)}
```

Comment:
Rather than input func better prop name will be onchange . Keep it in mind while writing next time

Reply:
I thought of not confusing it with a reserved keyword. Would consult more on this and will surely change if necessary.